packages/core/src/asciidoctor-core-api.js

```diff
@@ -3047,3 +3047,387 @@ Opal.Asciidoctor.SyntaxHighlighterBase = SyntaxHighlighterBase
 SyntaxHighlighterBase.prototype.registerFor = function (names) {
   SyntaxHighlighter['$register'](this, names)
 }
+
+// Table API
+
+/**
+ * Methods for managing AsciiDoc tables.
+ * @namespace
+ * @extends AbstractBlock
+ */
+var Table = Opal.Asciidoctor.Table
+
+/**
+ * Create a new Table element.
+ * @param {AbstractBlock} parent
+ * @param {Object|undefined} attributes
+ * @returns {Table}
+ */
+Table.create = function (parent, attributes) {
+  return this.$new(parent, toHash(attributes))
+}
+
+/**
+ * Get the caption of the table.
+ * @returns {string}
+ * @memberof Table
+ */
+Table.prototype.getCaption = function () {
+  return this.caption
+}
+
+/**
+ * Get the rows of this table.
+ * @returns {Object} An object with the members "head", "body" and "foot"
+ * @memberof Table
+ */
+Table.prototype.getRows = function () {
+  return this.rows
+}
+
+/**
+ * Get the columns of this table.
+ * @returns {Array<Column>}
+ * @memberof Table
+ */
+Table.prototype.getColumns = function () {
+  return this.columns
+}
+
+/**
+ * Get the head rows of this table.
+ * @returns {Array<Array<Cell>>} - an Array of Array of Cell
+ * @memberof Table
+ */
+Table.prototype.getHead = function () {
+  return this.rows.head
+}
+
+/**
+ * Get the body rows of this table.
+ * @returns {Array<Array<Cell>>} - an Array of Array of Cell
+ * @memberof Table
+ */
+Table.prototype.getBody = function () {
+  return this.rows.body
+}
+
+/**
+ * Get the foot rows of this table.
+ * @returns {Array<Array<Cell>>} - an Array of Array of Cell
+ * @memberof Table
+ */
+Table.prototype.getFoot = function () {
+  return this.rows.foot
+}
+
+/**
+ * Check if the table has rows.
+ * @returns {boolean}
+ * @memberof Table
+ */
+Table.prototype.hasRows = function () {
+  return this.rows !== Opal.nil && this.rows.body.length > 0
+}
+
+/**
+ * Check if the table has a header.
+ * @returns {boolean}
+ * @memberof Table
+ */
+Table.prototype.hasHeader = function () {
+  return this.rows !== Opal.nil && this.rows.head.length > 0
+}
+
+/**
+ * Check if the table has a header option set.
+ * @returns {boolean}
+ * @memberof Table
+ */
+Table.prototype.hasHeaderOption = function () {
+  return this.has_header_option
+}
+
+/**
+ * Check if the table has a footer.
+ * @returns {boolean}
+ */
+Table.prototype.hasFooter = function () {
+  return this.rows !== Opal.nil && this.rows.foot.length > 0
+}
+
+/**
+ * Check if the table has the footer option set.
+ * @returns {boolean}
+ * @memberof Table
+ */
+Table.prototype.hasFooterOption = function () {
+  var footerOption = this.getAttributes()['footer-option']
+  return footerOption === Opal.nil ? false : footerOption === ''
+}
+
+/**
+ * Check if the table has the autowidth option set.
+ * @returns {boolean}
+ * @memberof Table
+ */
+Table.prototype.hasAutowidthOption = function () {
+  var autowidthOption = this.getAttributes()['autowidth-option']
+  return autowidthOption === Opal.nil ? false : autowidthOption === ''
```

Comment:
@mojavelinux I'm always a bit concerned that someone could use `setAttribute` with a boolean value:

```js
table.setAttribute('autowidth-option', true)
table.hasAutowidthOption() // false?!
```

Do we need to check if the value is truthy instead? 

```suggestion
  return autowidthOption === Opal.nil ? false : autowidthOption
```